### PR TITLE
Add MariaDB Role (with galera clustering)

### PR DESCRIPTION
**Pull request by carsongee**
This primarily adds a new role for installing MariaDB for non-RDS instances and add active/active galera clustering as an option for running a load balanced and HA mysql cluster.  It borrowed quite a bit from edxlocal for user creation.

In addition to the new role I also made a cluster vagrant that stands up three boxes and installs all our cluster capable roles.  If you guys want me to split this in two, that is fine with me, but I used the cluster vagrant to help build the role so it ended up on this branch.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @carsongee! I've created [OSPR-164](https://openedx.atlassian.net/browse/OSPR-164) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams. Feel free to add information to the ticket which can help Product understand the context for the PR - supporting documentation, edx-code email threads, timeline information ('this must be merged by XX date', and why that is), partner information (this is for a course on edx.org, for example), etc. 

All technical communication about the code itself will still be done via the Github pull request interface. As a reminder, [our process documentation is here](http://edx.readthedocs.org/projects/userdocs/en/latest/process/overview.html).

File: playbooks/vagrant-cluster.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    mongo_create_users: yes
+    RABBITMQ_CLUSTERED: yes
+    MARIADB_CLUSTERED: yes
+    MARIADB_CREATE_DBS: yes
+  vars_files:
+    - "group_vars/all"
+    - "roles/analytics-api/defaults/main.yml"
+  roles:
+    - mongo
+    # - mariadb
```
Comment (e0d) [inline playbooks/vagrant-cluster.yml]: intended comment-out?

File: playbooks/roles/mariadb/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  apt: name={{ item }} update_cache=yes
+  with_items: mariadb_solo_packages
+  when: not MARIADB_CLUSTERED|bool
+
+- name: install mariadb cluster packages
+  apt: name={{ item }} update_cache=yes
+  with_items: mariadb_cluster_packages
+  when: MARIADB_CLUSTERED|bool
+
+- name: remove bind-address
```
Comment (e0d) [inline playbooks/roles/mariadb/tasks/main.yml:46]: why is this needed?

File: playbooks/roles/mariadb/handlers/main.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (e0d) [inline playbooks/roles/mariadb/handlers/main.yml]: If unneeded, remove.

File: playbooks/vagrant-cluster.yml (hunk 1 above)
Comment (carsongee) [inline playbooks/vagrant-cluster.yml]: woops, committed while debugging mongo, will fix

File: playbooks/roles/mariadb/tasks/main.yml (hunk 2 above)
Comment (carsongee) [inline playbooks/roles/mariadb/tasks/main.yml:46]: The cluster members need to be able to talk to each other as well, and it also allows the app to connect. Removing it sets it to all interfaces

File: playbooks/roles/mariadb/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+# Example play:
+#
+#
+
+- name: Install pre-req debian packages
+  apt: name={{ item }} state=present
+  with_items: mariadb_debian_pkgs
+
+- name: Add mongo key
+  apt_key: url="http://keyserver.ubuntu.com/pks/lookup?op=get&fingerprint=on&search={{ MARIADB_APT_KEY_ID }}"
```
Comment (e0d) [inline playbooks/roles/mariadb/tasks/main.yml]: We should probably pull the url up to common.  Also update the name here.

Comment (carsongee) [inline playbooks/roles/mariadb/tasks/main.yml]: want me to do this in this PR? how is `COMMON_UBUNTU_APT_KEYSERVER`?

**Comment by carsongee** [comment]
The elasticsearch cluster wasn't building due to vagrant always giving eth0 the same IP and elasticsearch (and mariadb) picking eth0 as their return address.  I added a `vagrant_cluster` var to common to tell both those apps to force their net addresses to the right ones and set it to true on the playbook.

File: playbooks/roles/mariadb/tasks/main.yml (hunk 4 above)
Comment (e0d) [inline playbooks/roles/mariadb/tasks/main.yml]: Name is good, not required but if not too disruptive it would be great.

**Comment by e0d** [comment]
LGTM at the moment, not sure if there are more changes planned or not.

File: playbooks/roles/mariadb/tasks/main.yml (hunk 4 above)
Comment (carsongee) [inline playbooks/roles/mariadb/tasks/main.yml]: no problem.  Looks like it is just here and in the dd role, so should be quick

**Comment by carsongee** [comment]
Just the move to the common key server url that I just committed.  Double check that for me and if you're good to merge, I am.

**Comment by e0d** [comment]
:+1:
